x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawl_detail_logic.test.ts

```diff
@@ -0,0 +1,132 @@
+  id: '12345',
+  status: CrawlerStatus.Pending,
+  created_at: 'Mon, 31 Aug 2020 17:00:00 +0000',
+  began_at: null,
+  completed_at: null,
+};
+
+const clientCrawlRequest = crawlRequestServerToClient(crawlRequestResponse);
+
+describe('CrawlDetailLogic', () => {
+  const { mount } = new LogicMounter(CrawlDetailLogic);
+  const { http } = mockHttpValues;
+  const { flashAPIErrors } = mockFlashMessageHelpers;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('has expected default values', () => {
+    mount();
+    expect(CrawlDetailLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('reducers', () => {
+    describe('closeFlyout', () => {
+      it('closes the flyout', () => {
+        mount({ flyoutClosed: false });
+
+        CrawlDetailLogic.actions.closeFlyout();
+
+        expect(CrawlDetailLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          flyoutClosed: true,
+        });
+      });
+    });
+
+    describe('fetchCrawlRequest', () => {
+      it('opens the flyout and sets loading to true', () => {
+        mount({
+          dataLoading: false,
+          flyoutClosed: true,
+        });
+
+        CrawlDetailLogic.actions.fetchCrawlRequest('12345');
+
+        expect(CrawlDetailLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          dataLoading: true,
+          flyoutClosed: false,
+        });
+      });
+    });
+
+    describe('onRecieveCrawlRequest', () => {
+      it('saves the crawl request and sets data loading to false', () => {
+        mount({
+          dataLoading: true,
+          request: null,
+        });
+
+        CrawlDetailLogic.actions.onRecieveCrawlRequest(crawlRequestResponse);
+
+        expect(CrawlDetailLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          dataLoading: false,
+          crawlRequestFromServer: crawlRequestResponse,
+          crawlRequest: clientCrawlRequest,
+        });
+      });
+    });
+  });
+
+  describe('listeners', () => {
+    describe('fetchCrawlRequest', () => {
+      it('updates logic with data that has been converted from server to client', async () => {
+        mount();
+        jest.spyOn(CrawlDetailLogic.actions, 'onRecieveCrawlRequest');
+
+        http.get.mockReturnValueOnce(Promise.resolve(crawlRequestResponse));
+
+        CrawlDetailLogic.actions.fetchCrawlRequest('12345');
+        await nextTick();
+
+        expect(http.get).toHaveBeenCalledWith(
+          '/internal/app_search/engines/some-engine/crawler/crawl_requests/12345'
+        );
+        expect(CrawlDetailLogic.actions.onRecieveCrawlRequest).toHaveBeenCalledWith(
+          crawlRequestResponse
+        );
+      });
+
+      it('displays any errors to the user', async () => {
+        mount();
+        http.get.mockReturnValueOnce(Promise.reject('error'));
+
+        CrawlDetailLogic.actions.fetchCrawlRequest('12345');
+        await nextTick();
+
+        expect(flashAPIErrors).toHaveBeenCalledWith('error');
+      });
```

Comment:
Totally optional, but you could use the helper I created here.

```suggestion
      itShowsServerErrorAsFlashMessage(http.get, () => {
        mount();
        CrawlDetailLogic.actions.fetchCrawlRequest('12345');
      });
```

Reply:
d'oh I forgot about this

Reply:
Done in a3efc7c